radix-engine-interface/src/blueprints/resource/non_fungible/non_fungible_resource_manager.rs

```diff
@@ -304,30 +290,61 @@ pub struct NonFungibleResourceManagerMintSingleRuidTypedInput<T> {
 pub type NonFungibleResourceManagerMintSingleRuidOutput = (Bucket, NonFungibleLocalId);
 
 #[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor, ManifestSbor)]
-pub struct NonFungibleDataSchema {
-    pub schema: VersionedScryptoSchema,
-    pub non_fungible: LocalTypeId,
-    pub mutable_fields: IndexSet<String>,
+pub enum NonFungibleDataSchema {
+    // TODO: ignore this variant in Scrypto for smaller code size
+    Local {
+        schema: VersionedScryptoSchema,
+        type_id: LocalTypeId,
+        mutable_fields: IndexSet<String>,
+    },
+    Remote {
+        type_id: BlueprintTypeId,
+        mutable_fields: IndexSet<String>,
+    },
 }
 
 impl NonFungibleData for () {
     const MUTABLE_FIELDS: &'static [&'static str] = &[];
 }
 
 impl NonFungibleDataSchema {
-    pub fn new_schema<N: NonFungibleData>() -> Self {
+    /// Arguments:
+    /// * [`package_address`] - The package address to use for resolving types defined within this package
```

Comment:
Is it worth saying that `None` are replaced with the current package?

Reply:
Done